File: pasta projeto_typescript_reflexao/versao-java-v1/aula01/OlaMundo.java
package aula01;

public class OlaMundo extends HelloWorld {
	public static void main(String[] args) {
		OlaMundo olaMundo = new OlaMundo();
		olaMundo.printName();
		olaMundo.teste(1,2);
		HelloWorld helloWorld = new HelloWorld();
		helloWorld.printName();
	}
	public void teste(int a, int b){
		System.out.println("Teste");
	}
	public void teste(int a, boolean b){

	}
}

File: pasta projeto_typescript_reflexao/projeto_typescript_reflexao/src/aula01/OlaMundo.ts
import { HelloWorld } from "./HelloWorld";

export class OlaMundo extends HelloWorld {

    private nomeFuncionalidade = "Eu sou uma funcionalidade";

    teste(a: number, b: number): void;
    teste(a: number, b: boolean): void;

    teste(a: number, b: number | boolean): void {
        console.log("Teste");
        console.log("a:", a);
        console.log("b:", b);
    }
}

